### ISSUE TITLE
git switch is not a command

### ISSUE THREAD

**Issue by BryceMehring** [JamesIves/github-pages-deploy-action#120]
**Describe the Issue**

Deploy fails with `git switch is not a command`

**Logs**

```
git switch develop
git: 'switch' is not a git command. See 'git --help'.
```
**Workflow** 

https://github.com/GoldbarGames/GoldbarGames.github.io/blob/develop/.github/workflows/ci.yml

**Comment by JamesIves** [comment]
I'm guessing this image has an outdated version of Git: https://github.com/GoldbarGames/GoldbarGames.github.io/blob/develop/.github/workflows/ci.yml#L10-L13

Can you try running `apt-get install git` in a separate job and see if this still fails with this error?  https://stackoverflow.com/questions/32212996/how-to-resolve-git-pull-is-not-a-git-command-in-linux

**Comment by JamesIves** [comment]
This is a pretty strange error because `git switch` is a [valid command](https://git-scm.com/docs/git-switch). I have a feeling it's to do with the use of the ruby container as that's the only difference between these workflows. 

Perhaps there's a version mismatch somewhere? If there's not an easy way to do that then it may be worth changing to the action so it uses `git checkout` instead of `git switch`.

**Comment by JamesIves** [comment]
With that being said; can you point your workflow to `releases/v3-test` and try re-running it? I made a change to the beta branch which eliminates the use of `git switch`.

**Comment by BryceMehring** [comment]
I'm now getting `There was an error when attempting to execute the process 'rsync'. This may indicate the process failed to start. Error: spawn rsync ENOENT`:

```
Running JavaScript Action with default external tool: node12
git init
Reinitialized existing Git repository in /__w/GoldbarGames.github.io/GoldbarGames.github.io/.git/
git config user.name BryceMehring
git config user.email brycemehring@gmail.com
git fetch
git ls-remote --heads https://***@github.com/GoldbarGames/GoldbarGames.github.io.git master | wc -l
62a34e0483b977043e3cbb855560c69cd9f5e6d0	refs/heads/master
git checkout --progress --force develop
Switched to a new branch 'develop'
Branch 'develop' set up to track remote branch 'develop' from 'origin'.
git fetch https://***@github.com/GoldbarGames/GoldbarGames.github.io.git
From https://github.com/GoldbarGames/GoldbarGames.github.io
 * branch              HEAD       -> FETCH_HEAD
git worktree add --checkout gh-action-temp-deployment-folder origin/master
Preparing worktree (detached HEAD 62a34e04)
HEAD is now at 62a34e04 test
rsync -q -av --progress _site/. gh-action-temp-deployment-folder --exclude .git --exclude .github
The deployment encountered an error. ❌
##[error]There was an error when attempting to execute the process 'rsync'. This may indicate the process failed to start. Error: spawn rsync ENOENT
```

**Comment by JamesIves** [comment]
Give this a try:

```yml
      - name: install, build, test
        run: |
          apt-get update && apt-get install -y rsync
          gem install bundler
          bundle install
          JEKYLL_ENV=production bundle exec jekyll build
          htmlproofer ./_site --check-html --allow-hash-href --disable-external
```

**Comment by BryceMehring** [comment]
The deploy now works!

**Comment by JamesIves** [comment]
Great! Thanks for testing that out. I've gone ahead and added a note about this in the README: https://github.com/JamesIves/github-pages-deploy-action/blob/issue-120/README.md#using-a-container- 

I'll likely push this up release up in the morning.

**Comment by JamesIves** [comment]
This change to `git switch` along with the the documentation has been updated and released as part of `3.1.2` I've also gone ahead and added an additional integration test that ensures that this workaround will continue to work should anything change with the ruby container.
